WordPress/src/main/java/org/wordpress/android/ui/posts/PostEditorAnalyticsSession.java

```diff
@@ -94,12 +96,15 @@ public static PostEditorAnalyticsSession getNewPostEditorAnalyticsSession(
         return new PostEditorAnalyticsSession(editor, post, site, isNewPost);
     }
 
-    public void start(ArrayList<Object> unsupportedBlocksList) {
+    public void start(ArrayList<Object> unsupportedBlocksList, GutenbergPropsBuilder capabilities) {
         if (!mStarted) {
             mHasUnsupportedBlocks = unsupportedBlocksList != null && unsupportedBlocksList.size() > 0;
             Map<String, Object> properties = getCommonProperties();
             properties.put(KEY_UNSUPPORTED_BLOCKS,
                     unsupportedBlocksList != null ? unsupportedBlocksList : new ArrayList<>());
+            if (capabilities != null) {
+                properties.put(KEY_CAN_VIEW_EDITOR_ONBOARDING, capabilities.getCanViewEditorOnboarding());
```

Comment:
Is there a more appropriate approach to accessing this value rather than importing the `GutenbergPropsBuilder` module and making this value public?

Reply:
Since you're not making anythign in the `GutenbergPropsBuilder` mutable, I think doing it this way is fine. 

With that said, this is not a blocker, but I would do this differently and just pass the relevant boolean value to the `PostEditorAnalyticsSession` instead of the entire `GutenbergPropsBuilder` object. It feels to me like this is kind of using the `GutenbergPropsBuilder` for a purpose (passing information to this class) that it was not designed for (i.e., we wouldn't want someone in the future to add something to the `GutenbergPropsBuilder` just in order to use it in this method). This is admittedly a bit of a matter of style, so I do not consider this a blocker and you can leave this as-is if you prefer, but I would do something like this instead: https://github.com/wordpress-mobile/WordPress-Android/commit/ce5a6245ecb947527a88f6ea2075473bccc2a334.

Reply:
Thanks, @mchowning! This is exactly the type of feedback for which I was looking. I am less familiar with Java, so I welcome the ideas. 

> With that said, this is not a blocker, but I would do this differently and just pass the relevant boolean value to the `PostEditorAnalyticsSession` instead of the entire `GutenbergPropsBuilder` object. 

This makes total sense, particularly with the rationale you provided. Passing the entirety of capabilities as the parameter is a leftover from my original implementation where I serialized the entire `capabilities` as a stringified JSON object. However, that value is less valuable in Tracks, as we cannot easily segment on a single capability. 

So, I agree with your changes in https://github.com/wordpress-mobile/WordPress-Android/commit/ce5a6245ecb947527a88f6ea2075473bccc2a334. If you don't mind, I'll likely include them. It may even make sense to update the iOS implementation to mimic its structure as well.

Reply:
Refactored this to rely upon a recommended boolean value in https://github.com/wordpress-mobile/WordPress-Android/pull/14766/commits/b690fcd5b738667e9140c49901ee80b154cc0111.